### PR TITLE
Exclude param

### PR DESCRIPTION
**Pull request by matteofigus**
## Overview

Adds an exclude param as discussed in #68 
At the moment, it supports normalized and non-normalized paths, and wildcards such as 
`--exclude foo` or `--exclude ./foo`, `--exclude *.txt`. It also supports multiple `--exclude` params.

NOTE:
In the issue, we discussed about both adding a parameter and a .taskcatignore file for similar behavior. I decided I'll work on the .taskcatignore change separately to keep the PR small - much of what's here can definitely be reused for that.

## Testing/Steps taken to ensure quality

* Manually tested with https://github.com/aws-samples/amazon-rekognition-engagement-meter

### CODE REVIEW

File: taskcat/s3_sync.py [hunk 1]
```diff
@@ -22,33 +22,37 @@
...
+
+    exclude_defaults = [
+        hidden_files_and_folders,
+        lambdas_source,
         "*.md"
     ]
-    exclude_path_prefixes = [
-        "functions/source/",
-        "."
-    ]
```
Comment (agamzn) [inline taskcat/s3_sync.py]: We leverage this class for other toolsets and depend on the current class attributes (including this one). If possible - can you not modify the S3Sync class and instead set the attributes prior to uploading?  Ex:

```
S3Sync.exclude_path_prefixes.append(...)
S3Sync(...)
```

Comment (agamzn) [reply taskcat/s3_sync.py]: /cc @jaymccon

Comment (matteofigus) [reply taskcat/s3_sync.py]: ok done

File: taskcat/stacker.py [hunk 2]
```diff
@@ -1550,6 +1560,12 @@ def interface(self):
             '--lambda-build-only',
             action='store_true',
             help="create lambda zips and exit")
+        parser.add_argument(
+            '-e',
+            '--exclude',
+            action='append',
+            help="Exclude directory or files from s3 sync"
```
Comment (agamzn) [inline taskcat/stacker.py]: Minor adjustment, can you edit this text to include a brief example?

```
--exclude foo --exclude bar --exclude other
```

**Pull request merged**